File: software/examples/perl/example_input.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletIO16;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your IO-16 Bricklet

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $io = Tinkerforge::BrickletIO16->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Get current value from port A as bitmask
my $value_mask_a = $io->get_port('a');
print "Value Mask (Port A): " . sprintf('%08b', $value_mask_a) . "\n";

# Get current value from port B as bitmask
my $value_mask_b = $io->get_port('b');
print "Value Mask (Port B): " . sprintf('%08b', $value_mask_b) . "\n";

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();

File: software/examples/ruby/example_input.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_io16'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your IO-16 Bricklet

ipcon = IPConnection.new # Create IP connection
io = BrickletIO16.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Get current value from port A as bitmask
value_mask = io.get_port 'a'
puts "Value Mask (Port A): #{'%08b' % value_mask}"

# Get current value from port B as bitmask
value_mask = io.get_port 'b'
puts "Value Mask (Port B): #{'%08b' % value_mask}"

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
